org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/XMLWorkspaceService.java

```diff
@@ -54,4 +80,33 @@ public void didChangeWatchedFiles(DidChangeWatchedFilesParams params) {
 			}
 		}
 	}
+
+	@Override
+	public void registerCommand(String commandId, Function<ExecuteCommandParams, CompletableFuture<Object>> handler) {
+		synchronized (commands) {
+			if (commands.containsKey(commandId)) {
+				throw new IllegalArgumentException("Command with id '" + commandId + "' is already registered");
+			}
+			String registrationId = UUID.randomUUID().toString();
+			xmlLanguageServer.capabilityManager.registerCapability(registrationId, WORKSPACE_EXECUTE_COMMAND, ImmutableMap.of("commands", ImmutableList.of(commandId)));

```

Comment:
Please update existing code to define a getCapabilityManager() and set the capabilityManager as private. Thanks

Reply:
Fixed